src/common/mem.c

```diff
@@ -38,23 +38,34 @@ enum {
 	DAOS_MD_PMEM	= 0,
 	DAOS_MD_BMEM	= 1,
 	DAOS_MD_ADMEM	= 2,
+	DAOS_MD_INVAL	= 4,
 };
 
 static int daos_md_backend = DAOS_MD_PMEM;
 
-/** Sets up global settings for the pmem objects.
+/** Initializes global settings for the pmem objects.
+ *
+ *  \param	md_on_ssd[IN]	Boolean indicating if MD-on-SSD is enabled.
  *
  *  \return	0 on success, non-zero on failure.
  */
 int
-umempobj_settings_init(void)
+umempobj_settings_init(bool md_on_ssd)
 {
 	int					rc;
 	enum pobj_arenas_assignment_type	atype;
-	unsigned int				val = DAOS_MD_PMEM;
+	unsigned int				md_mode = DAOS_MD_INVAL;
+
+	d_getenv_int("DAOS_MD_ON_SSD", &md_mode);
+
+	if (md_on_ssd && md_mode == DAOS_MD_PMEM) {
+		D_ERROR("MD on SSD mode detected but DAOS_MD_ON_SSD set to MD_PMEM (0)");
+	}
 
-	d_getenv_int("DAOS_MD_ON_SSD", &val);
-	switch (val) {
+	/* Use input argument (mode detected) if not set explicitly by env var */
+	md_mode = (md_mode == DAOS_MD_INVAL) ? (int)md_on_ssd : md_mode;
```

Comment:
This looks not correct to me. If md-on-ssd is configured but the env var isn't set, we should use default "BMEM", right? I think it could be like:
1. If md-on-ssd isn't configured, used "PMEM";
2. If md-on-ssd is configured, use the the env var specified backend, if env var isn't specified, use default "BMEM";
3. The env var accepts only two backends, "BMEM" or "ADMEM";

Reply:
That is what will happen? If env not set then md_mode is set to md_on_ssd (if enabled that will be BMEM, otherwise PMEM).  
If env is set then whichever mode specified will be used.
I will make your changes anyway.

Reply:
Why do we need to override config by setting env var to PMEM? What the use case? It looks dangerous to me.

To exercise md-on-ssd in vos_tests, you need to prepare NVMe devices, when you look into the run_test.sh, you'll see that we used AIO devices for vos_tests to exercise md-on-ssd.

Reply:
Thanks for the explanations, suggested changes made. Done